### ISSUE TITLE
WARN on UPLOAD with No name found in C:\devAC\cus\xtk_javascript\racc\Connector.j

### ISSUE THREAD

**Issue by ScyriusS** [cumpstey/campaign-sync#2]
Hello,

I extracted all my custom javascript librairies thank to you tools, this part works very fine. 👍 

BUT, when i want to import in other instance, i have a WARN and no import done, please see the log below :

```
> WARN  2018-10-05 09:52:17 - No name found in C:\devAC\racc\xtk_javascript\cus\CampaignBuilder.js.
> WARN  2018-10-05 09:52:17 - No name found in C:\devAC\racc\xtk_javascript\cus\Connector.js.
> WARN  2018-10-05 09:52:17 - No name found in C:\devAC\racc\xtk_javascript\cus\DatabaseOperations.js.
> WARN  2018-10-05 09:52:17 - No name found in C:\devAC\racc\xtk_javascript\cus\RestAPI.js.
```

**I didn't fine in the source code where is generated this error and why.**

Thank for any help.

**Comment by Rogn** [comment]
Hi,
Can you show how the metadata header of one of those files looks like?
It will look something like this:
/*!
Adobe Campaign metadata:
Schema: xtk:javascript
Name: cus:Footerxml2Json.js
Label: FooterXml2Json
!*/

/Hallur

**Comment by Rogn** [comment]
But from the headline of your Issue, I can guess that it is related to the namespace ('racc'), which you are using.
That is used as part of the name. And when the Name gets reread from the file metadata, the following parsing rules get applied:
The first part of the namespace e.g. 'cus' can only contain letters from a-z and be 3 characters long. And the name part e.g. 'Footerxml2Json.js' can only contain letters from a-z, numbers 0-9 and the letters '.' and '_'. The case is not important.
And in your case it looks like your namespace part '**racc**:Connector.js' is 4 letters long.

/Hallur

**Comment by cumpstey** [comment]
Yes, it sounds like an issue with the namespace being unexpectedly _not_ three characters. Thanks @Rogn.

In the version of Campaign I was using when I wrote this, the namespace was restricted to being three characters. There may well be other "works with my instance" issues along similar lines, given that I've only ever worked on the one project with Campaign.

I'm now basically unable to support this project as I'm no longer working with Campaign, but if anyone would like to submit a pull request to allow longer namespaces, I'll merge it in; or I may eventually get round to changing it myself, but likely not soon.

**Comment by ScyriusS** [comment]
Thank you for your answer / i create a barcnh of your project on git so , i
will try to implement an evolution on this point.

FYI, I analyse to do evolution to manage report and delivery schemas. i
will commit asap on github.

Thank you for your support

Le lun. 8 oct. 2018 à 10:41, Neil Cumpstey <notifications@github.com> a
écrit :

**Comment by ScyriusS** [comment]
Hello All,

my knowledge on C# is very limited and i didn't found where i must update the code to manage namespace with more of 3 characters. Maybe one body can help me ? 

The goal is use campaignSync to manage specific module with a namespace must have 5 characters.

Thank for anay help..

Alex

**Comment by cumpstey** [comment]
I've just committed a change on the `feature/issue-2` branch, which I think should allow the namespace to be any length (still letters only). I think this should be the only change necessary, but I have no way of testing at the moment - so could you please try this out and see whether it works?
